person.rb

```diff
@@ -0,0 +1,44 @@
+require './nameable'
+require './capitalizedecorator'
+require './trimmerdecorator'
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id, :rental
+
+  def can_use_services?
+    return unless of_age? || @parent_permission
+
+    true
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```

Comment:
**1-** Moreover, for the add_rental method The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```
 def add_rental(book, date)
    Rental.new(date, book, self)
  end
```